README.adoc

```diff
@@ -1,309 +1,49 @@
 = meta-updater
 :toc: macro
 :toc-title:
+:meta-updater-github-url: https://github.com/advancedtelematic/meta-updater/blob/master/
```

Comment:
I wish there was a way to get this to point at whatever the current branch is instead of always assuming master. @tkfu is that asking for too much?

Reply:
That's possible, yes, though it's mildly ugly. You'd need these lines in place of the line above in each file that uses it:

```
ifdef::env-github[:meta-updater-github-url: ../../../../]
ifndef::env-github[:meta-updater-github-url: https://github.com/advancedtelematic/meta-updater/blob/master/]
```

Reply:
@merltron This also gets at some important IA questions surrounding versioning. Antora's conceptual model of "components" is designed to make it easy to have docs affecting a specific component get versioned. However, this depends on having the docs structured in a way that the docs for each component are more or less independent. Right now, that's not generally the case; the developer guide has content that could potentially change based on the meta-updater version and/or the aktualizr version, including many files that talk about the different ways to set options in the sota.toml file vs. yocto's local.conf.

I think that overall that's probably the right decision, but it does mean that our versioning strategy, once we sort it out better, will mean that there will have to be a version option in the docs for each possible tuple of aktualizr_version and meta_updater_version.

@patrickvacek Let's have a meeting about versioning/tagging plans going forward? I think in my ideal world, every time we made a release of aktualizr the embedded team would decide which branches they wanted to backport it to, and then tag a commit on each branch with something like `rocko-2019.5`. (Also in this ideal world, no branch but master would ever be building an aktualizr that isn't a point release.)

Reply:
> That's possible, yes, though it's mildly ugly.

Do you have a strong preference here? I'll generally default to you and Merlin, but I'm still gonna express my opinions. :)

> Let's have a meeting about versioning/tagging plans going forward?

Sure, sounds like a good idea.

> I think in my ideal world, every time we made a release of aktualizr the embedded team would decide which branches they wanted to backport it to, and then tag a commit on each branch with something like `rocko-2019.5`.

That sounds like a reasonable strategy. However, I would posit that every release should be backported to every Yocto release we claim to actively support. To make that easier, we should try to limit how many releases we support, and that why we should drop rocko/sumo ASAP.

We might also want to try to automate the release process to simplify this somewhat. There's a ticket for that work, but it hasn't been a high priority since it looks like a fair bit of work for not much gain. But if we want to release more often, it might be worth it.

> Also in this ideal world, no branch but master would ever be building an aktualizr that isn't a point release.

That also sounds reasonable.

Reply:
> I think that overall that's probably the right decision, but it does mean that our versioning strategy, once we sort it out better.

@tkfu : Yes, but when Patrick raised the topic of versioning, I did have a consider having a separate guide about the "OTA Connect Yocto Layer" or more generally the "OTA Connect Build Tools". It's just a challenge to give people some of the more general conceptual information. Once solution is to treat some "components" more as "Chapters", so that you're using the top nav to cycle through the different chapters.  (ie: most conceptual content doesn't need versioning) Its just a challenge as to how to slice it up. But a meeting is a good idea.

Reply:
> That's possible, yes, though it's mildly ugly. You'd need these lines in place of the line above in each file that uses it:
> 
> ```
> ifdef::env-github[:meta-updater-github-url: ../../../../]
> ifndef::env-github[:meta-updater-github-url: https://github.com/advancedtelematic/meta-updater/blob/master/]
> ```
@tkfu 
Why is this going 4 levels up if the "env" is Github? - the README is already in the root of the repo. I originally used just relative path, but Github seems to want `blob/<branch>/` to be in the path.

Reply:
Ah, you're right. I didn't notice this was the top-level README--I thought it was part of the content catalog.

@patrickvacek Unfortunately, for READMEs, there's no good way to do relative links on github. The most likely case is that the reader is viewing it from https://github.com/advancedtelematic/meta-updater, but it's also possible that they're viewing it from https://github.com/advancedtelematic/meta-updater/blob/master/README.adoc. We can't even distinguish between those two cases. Add in the possibility of https://github.com/advancedtelematic/meta-updater/tree/{branchname} and https://github.com/advancedtelematic/meta-updater/blob/{branchname}/README.adoc and it's even more hairy.

Even if we don't consider the possibility of someone reading via the `/blob/` urls, there's no relative link that works in the https://github.com/advancedtelematic/meta-updater/tree/{branchname} case, because we need to replace 'tree' with 'blob' in the link, and there's no URL rewriting logic available to us in the github renderer.

Reply:
Updated to use "tree" instead of "blob" which seems to work better.

Reply:
I think they're basically equivalent nowadays...github will redirect tree to blob when it's a file, and redirect blob to tree when it's a directory (both of those cases used to be 404s). It's just that we still don't have a solution for the most common case, where the source page is the main page of the project and there is no branch specified in the URL.